lib/fennec/evaluator/allocate/request.ex

```diff
@@ -0,0 +1,107 @@
+defmodule Fennec.Evaluator.Allocate.Request do
+  @moduledoc false
+
+  alias Jerboa.Format.Body.Attribute
+  alias Jerboa.Params
+  alias Fennec.TURN
+  @lifetime 10 * 60
+
+  @spec service(Params.t, map, TURN.t) :: {Params.t, TURN.t}
+  def service(params, changes, turn_state) do
+    request_status =
+      {:continue, params, %{}}
+      |> maybe(&verify_existing_allocation/4, [changes, turn_state])
+      |> maybe(&verify_requested_transport/2)
+      |> maybe(&verify_unknown_attributes/2)
+
+    case request_status do
+      {:error, error_code} ->
+        {%{params | attributes: [error_code]}, turn_state}
+      {:respond, {new_params, new_turn_state}} ->
+        {new_params, new_turn_state}
+      {:continue, _, state} ->
+        allocate(params, state, changes, turn_state)
+    end
+  end
+
+  defp allocate(params, _state, changes, turn_state) do
+    addr = Application.get_env(:fennec, :relay_addr, {127, 0, 0, 1})
+    {:ok, socket} = :gen_udp.open(0, [:binary, active: true, ip: addr])
+    allocation = %Fennec.TURN.Allocation{
+      socket: socket,
+      expire_at: System.system_time(:second) + @lifetime,
+      owner: Params.get_id(params)
```

Comment:
It's not. The owner is the transaction that created the Allocation. This is only to make this request idempotent. The RFC says, that if the server receives the same request twice (same transaction id), which may happen if the client sends packets with redundancy, the server should not report an error. Instead, it should respond as it would be the original allocation, only with slightly different time-to-live of the allocation (because of the time difference between those requests).